backend-server/shared/starters/oss/src/main/java/com/apitable/starter/oss/core/obs/HuaweiCloudOssClientRequest.java

```diff
@@ -0,0 +1,309 @@
+package com.apitable.starter.oss.core.obs;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.Serializable;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.List;
+import java.net.HttpURLConnection;
+import java.util.function.Consumer;
+
+import com.apitable.starter.oss.core.OssStatObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.apitable.starter.oss.core.AbstractOssClientRequest;
+import com.apitable.starter.oss.core.OssObject;
+import com.apitable.starter.oss.core.UrlFetchResponse;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.json.JsonMapper;
+import com.obs.services.ObsClient;
+import com.obs.services.exception.ObsException;
+import com.obs.services.model.AbortMultipartUploadRequest;
+import com.obs.services.model.CompleteMultipartUploadRequest;
+import com.obs.services.model.InitiateMultipartUploadRequest;
+import com.obs.services.model.InitiateMultipartUploadResult;
+import com.obs.services.model.ObjectMetadata;
+import com.obs.services.model.ObsObject;
+import com.obs.services.model.PartEtag;
+import com.obs.services.model.PutObjectResult;
+import com.obs.services.model.StorageClassEnum;
+import com.obs.services.model.UploadPartRequest;
+import com.obs.services.model.UploadPartResult;
+
+import cn.hutool.core.io.IoUtil;
+import cn.hutool.core.util.StrUtil;
+import cn.hutool.core.util.URLUtil;
+
+public class HuaweiCloudOssClientRequest  extends AbstractOssClientRequest{
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(HuaweiCloudOssClientRequest.class);
+  
+    private final ObsClient obsClient;
+    private boolean autoCreateBucket = false;
+    private MultipartState multipartState;
```

Comment:
convert local variable since this variable used in `uploadStreamForObject` method only